File: GEMP - Iniciação a Estrutura de Dados/SegredoDoCofre.cpp
#include <iostream>

using namespace std;

int mCount[100001][10];
int ans[10];

int main() {
    int n = 0, m = 0;
    int aux = 0, start = 0, end = 0;
    
    cin >> n >> m;
    
    for(int i = 1; i <= n; i++) {
        cin >> aux;
        for(int j = 0; j < 10; j++) {
            mCount[i][j] = mCount[i-1][j];
        }
        mCount[i][aux]++;
    }

    cin >> end;

    for(int i = 1; i < m; i++) {
        cin >> end;
        aux = end;
        
        if(start > end) {
            start--;
            end--;
        }
        
        for(int j = 0; j < 10; j++) {
            ans[j] += abs(mCount[end][j] - mCount[start][j]);
        }
        
        start = aux;
    }
    
    for(int i = 0; i < 10; i++) {
        cout << ans[i] << " ";
    }
    
    cout << endl;

    return 0;
}

File: ProgramacaoIntermediaria/SegredoDoCofre.py
n,m = input().split()
vetor = input().split()
consultas = input().split()
prefix = [[0,0,0,0,0,0,0,0,0,0]]

for i in range(int(n)):
    j = int(vetor[i])
    lista = prefix[i].copy()
    lista[j]+=1
    prefix.append(lista)

ans = [0,0,0,0,0,0,0,0,0,0]
start = int(consultas[0])-1

for i in range(1,int(m)):
    end = int(consultas[i])
    aux = end
    
    if end < start:
        end-=1
        start-=1
    
    for j in range(10):
        ans[j] += abs(prefix[end][j] - prefix[start][j])
    
    start = aux

for i in range(9):
    print(ans[i], end = " ")
print(ans[9])